x-pack/plugins/secret_service/server/service/secret_service.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignores
+import nodeCrypto from '@elastic/node-crypto';
+import crypto from 'crypto';
+import { SavedObjectsClient } from 'src/legacy/server/saved_objects';
+// @ts-ignore
+import { AuditLogger } from '../../../server/lib/audit_logger';
+
+type SSEvents = 'secret_object_created' | 'secret_object_accessed' | 'secret_object_decrypt_failed';
+type SecretId = Promise<string>;
+
+export class SecretService {
+  public readonly hideAttribute: (deets: any) => SecretId;
+  public readonly unhideAttribute: (deets: any) => any;
+  public readonly validateKey: () => Promise<boolean>;
+  private readonly type: string;
+  private readonly auditor?: AuditLogger;
+  private readonly hideAttributeWithId: (id: string, toEncrypt: any) => SecretId;
+
+  constructor(
+    savedObjectsClient: SavedObjectsClient,
+    type: string,
+    key?: Buffer,
+    auditor?: AuditLogger
+  ) {
+    this.type = type;
+    this.auditor = auditor;
+    key = key || crypto.randomBytes(128);
+    const crypt = nodeCrypto({ encryptionKey: key.toString('hex') });
```

Comment:
It is a buffer not a string, the `toString('hex')` converts it to a string in hex format from the buffer representation. But since we are only allowing valid values we can probably just request the string representation in the first place.

Reply:
https://github.com/njd5475/kibana/blob/alerting/needed-services/x-pack/plugins/secret_service/index.ts#L47

Reply:
and AFAIK `JSON.parse` doesn't ever return buffers, so I'm not sure how `keystore.get` would ever return anything but a string.

Reply:
I think I see where the confusion is, I made these comments with the assumption that we'd be getting rid of `key = key || crypto.randomBytes(128);` since it's always provided.

Reply:
Yeah so I am removing that and will also remove the `.toString('hex')`